pkg/controllers/statusaggregator/controller.go

```diff
@@ -347,19 +343,60 @@ func (a *StatusAggregator) reconcile(ctx context.Context, key reconcileKey) (sta
 		return worker.StatusError
 	}
 
-	newObj, needUpdate, err := plugin.AggregateStatuses(ctx, sourceObject, fedObject, clusterObjs, clusterObjsUpToDate)
-	if err != nil {
-		return worker.StatusError
+	needUpdate := false
+	for _, plugin := range plugins.DefaultCommonPlugins {
+		newObj, newNeedUpdate, err := plugin.AggregateStatuses(ctx, sourceObject, fedObject, clusterObjs, clusterObjsUpToDate)
+		if err != nil {
+			return worker.StatusError
+		}
+		sourceObject = newObj
+		needUpdate = needUpdate || newNeedUpdate
 	}
 
 	if needUpdate {
-		logger.V(1).Info("Updating status of source object")
+		logger.V(1).Info("Updating metadata of source object")
 		_, err = a.dynamicClient.Resource(ftc.GetSourceTypeGVR()).Namespace(key.namespace).
-			UpdateStatus(ctx, newObj, metav1.UpdateOptions{})
+			Update(ctx, sourceObject, metav1.UpdateOptions{})
 		if err != nil {
 			if apierrors.IsConflict(err) {
 				return worker.StatusConflict
 			}
+			return worker.StatusError
+		}
+	}
+
+	needUpdate = false
+	plugin := plugins.GetPlugin(key.gvk)
+	if plugin != nil {
+		newObj, newNeedUpdate, err := plugin.AggregateStatuses(ctx, sourceObject, fedObject, clusterObjs, clusterObjsUpToDate)
+		if err != nil {
+			return worker.StatusError
+		}
+		sourceObject = newObj
+		needUpdate = needUpdate || newNeedUpdate
+	} else {
+		logger.V(3).Info("Plugin not found")
+	}
+
+	if needUpdate {
+		if err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
Why do we choose to retry on conflict here? A change in the source object might cause the status aggregated in this loop to become outdated

Reply:
The original concern here is that the Update action has just been executed before, and I am afraid that the cache will not be refreshed in time and cause conflicts. After careful inspection, the cache is not used here. Updated.